rasa/cli/export.py

```diff
@@ -0,0 +1,508 @@
+        if max_timestamp < min_timestamp:
+            cli_utils.print_error_and_exit(
+                f"Maximum timestamp '{max_timestamp}' is smaller than minimum "
+                f"timestamp '{min_timestamp}'. Exiting."
+            )
+
+
+def _ensure_pika_channel_is_open(
+    event_broker: PikaEventBroker,
+    attempts: int = 1000,
+    wait_time_between_attempts: float = 0.01,
+) -> None:
+    """Spin until the pika channel is open.
+
+    Prints an error and exits if the channel does not open.
+
+    It typically takes 50 ms or so for that to happen. We'll wait up to 10
+    seconds just in case.
+
+    Args:
+        event_broker: Pika event broker to wait for.
+        attempts: Number of retries.
+        wait_time_between_attempts: Wait time between retries.
+
+    """
+    while attempts:
+        if event_broker.channel:
+            return
+        time.sleep(wait_time_between_attempts)
+        attempts -= 1
+
+    cli_utils.print_error_and_exit("Failed to open Pika channel. Exiting.")
+
+
+def _prepare_pika_producer(event_broker: EventBroker) -> None:
+    """Sets `should_keep_unpublished_messages` flag to `False` if `event_broker`
+    is a `PikaEventBroker`.
+
+    If publishing of events fails, the `PikaEventBroker` instance should not keep a
+    list of unpublished messages, so we can retry publishing them. This is because
+    the instance is launched as part of this short-lived export script, meaning the
+    object is destroyed before it might be published.
+
+    In addition, wait until the `channel` property is available.
+
+    Args:
+        event_broker: Event broker to modify if it's a `PikaEventBroker`.
+
+    """
+    if isinstance(event_broker, PikaEventBroker):
+        event_broker.should_keep_unpublished_messages = False
+        event_broker.raise_on_failure = True
+
+        # the pika channel takes a short while to open, so we'll spin until that happens
+        _ensure_pika_channel_is_open(event_broker)
+
+
+def export_trackers(args: argparse.Namespace) -> None:
+    """Export events for a connected tracker store using an event broker.
+
+    Args:
+        args: Command-line arguments to process.
+
+    """
+    _inspect_timestamp_options(args)
+
+    endpoints = _get_available_endpoints(args.endpoints)
+    rasa_tracker_store = _get_rasa_tracker_store(endpoints)
+    event_broker = _get_event_broker(endpoints)
+    _prepare_pika_producer(event_broker)
+
+    requested_conversation_ids = _get_requested_conversation_ids(args.conversation_ids)
+    conversation_ids_to_process = _get_conversation_ids_to_process(
+        rasa_tracker_store, requested_conversation_ids
+    )
+
+    _publish_events(
+        rasa_tracker_store,
+        event_broker,
+        conversation_ids_to_process,
+        args.minimum_timestamp,
+        args.maximum_timestamp,
+        args.endpoints,
+        requested_conversation_ids,
+    )
+
+
+def _sort_and_select_events_by_timestamp(
```

Comment:
how about moving the actual "migration code" to a separate module?

Reply:
I don't think we need to create another module - this isn't used elsewhere

Reply:
It's less about the usage from other things, but more about separating the CLI part from the logic handling the actual migration. However, consider this optional.

Reply:
migrator was moved to new module